sphinx/writers/html.py

```diff
@@ -481,17 +482,25 @@ def visit_image(self, node):
             # Try to figure out image height and width.  Docutils does that too,
             # but it tries the final file name, which does not necessarily exist
             # yet at the time the HTML file is written.
-            if Image and not ('width' in node and 'height' in node):
+            if not ('width' in node and 'height' in node):
                 try:
-                    im = Image.open(os.path.join(self.builder.srcdir, olduri))
+                    filename = os.path.join(self.builder.srcdir, olduri)
+                    size = imagesize.get(filename)
+                    if size[0] == -1 and Image:  # fallback to PIL
+                        im = Image.open(os.path.join(self.builder.srcdir, olduri))
+                        size = im.size
```

Comment:
I think it would be nice if image related code is isolated into dedicated module.